src/TokenRunner/DocBlock/MalformWorker/ParamNameTypoMalformWorker.php

```diff
@@ -93,20 +95,36 @@ private function getAnnotationsOfType(string $docContent, string $type): array
 
     /**
      * @param string[] $argumentNames
+     * @param string[] $missArgumentNames
      * @param string[] $paramNames
      */
-    private function fixTypos(array $argumentNames, array $paramNames, string $docContent): string
+    private function fixTypos(array $argumentNames, array $missArgumentNames, array $paramNames, string $docContent): string
     {
-        foreach ($argumentNames as $key => $argumentName) {
+        // A table of permuted params. initialized by $argumentName instead of $paramNames is correct
+        $replacedParams = array_fill_keys($argumentNames, false);
+
+        foreach ($missArgumentNames as $key => $argumentName) {
             // 1. the same position
             if (! isset($paramNames[$key])) {
                 continue;
             }
 
             $typoName = $paramNames[$key];
-            $replacePattern = '#@param(.*?)' . preg_quote($typoName, '#') . '#';
+            $replacePattern = '#@param(.*?)(' . preg_quote($typoName, '#') . '\b)#';
```

Comment:
Add `\b` to make the pattern for `$foo` not match `$fooo`.

https://www.php.net/manual/en/regexp.reference.escape.php#:~:text=%5Cb-,word%20boundary,-%5CB